### PR TITLE
Pass sort for agg multiple

### PR DESCRIPTION
**Pull request by TomAugspurger**
xref https://github.com/pandas-dev/pandas/issues/20909

### CODE REVIEW

**Comment by TomAugspurger** [comment]
cc @jorisvandenbossche

File: pandas/tests/frame/test_apply.py [hunk 1]
```diff
@@ -908,6 +908,29 @@ def test_demo(self):
...
+                                 "C": ['bar', 'foobarbaz'],
+                                 "D": [pd.Timestamp('2013-01-01'), pd.NaT]},
+                                index=['min', 'sum'])
+        # sorted index
+        with tm.assert_produces_warning(None):
+            result = mdf.agg(['min', 'sum'])
+
+        tm.assert_frame_equal(result, expected)
+
+        # unsorted index. Still sorts the index for backwards compat
```
[the remark this answers is not in the document]
Comment (TomAugspurger) [reply pandas/tests/frame/test_apply.py]: Even though we pass `.agg(['sum', 'min'])` the result's index is the *sorted* function names, `['min', 'sum']`. Make sense?

[the remark this answers is not in the document]
Comment (TomAugspurger) [reply pandas/tests/frame/test_apply.py]: I could imaging preserving the function order here someday, but would need to deprecate first.

[the remark this answers is not in the document]
Comment (TomAugspurger) [reply pandas/tests/frame/test_apply.py]: I can update the comment to say "Still shorts the *result's* index"

**Comment by TomAugspurger** [comment]
CI passed. Pushing a clarifying comment and merging.

File: pandas/tests/frame/test_apply.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jorisvandenbossche) [reply pandas/tests/frame/test_apply.py]: ah, ok, the sorting is about the [sum, min] axis, then I understand

**Pull request merged**